embassy-stm32/src/usart/mod.rs

```diff
@@ -359,6 +359,79 @@ mod eh1 {
     impl<'d, T: BasicInstance, RxDma> embedded_hal_1::serial::ErrorType for UartRx<'d, T, RxDma> {
         type Error = Error;
     }
+
+    
+    impl<'d, T: BasicInstance, RxDma> embedded_hal_1::serial::nb::Read for UartRx<'d, T, RxDma> {
+        fn read(&mut self) -> nb::Result<u8, Self::Error> {
```

Comment:
could this call the `eh02` code, to not copypaste? or better, make it an inherent private fn and call it from trait impls, this way we don't have any logic in the trait impls.